refactor(core): update undecorated-classes-with-di migration rerun command

alan-agius4 opened the pull request:
With Angular CLI version 9 RC 3 we can run a single migration for a package using the name of the migration schematic.

We need to pass the schematic name as a value to the `migrate-only` option.
Ex:
```
ng update @angular/core --migrate-only migration-v9-undecorated-classes-with-di
```

See: https://github.com/angular/angular-cli/pull/16174


## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

devversion reviewed (commented):
Nice. That is a very useful feature :100: For this transition phase though, can we be sure that people have the latest CLI version installed? I feel like we cover more cases/versions with the old command right now? Nothing wrong with re-running the other migrations as a noop.

devversion commented:
Also, is this really a fix? This looks more like it would fit into the `refactor` scope.

alan-agius4 commented:
During `ng update` the latest CLI is always installed.

Updated the commit msg

alan-agius4 committed:
refactor(core): update undecorated-classes-with-di migration rerun command

With Angular CLI version 9 RC 3 we can run a single migration for a package using the name of the migration schematic.

We need to pass the schematic name as a value to the `migrate-only` option.
Ex:
```
ng update @angular/core --migrate-only migration-v9-undecorated-classes-with-di
```

See: https://github.com/angular/angular-cli/pull/16174

devversion commented:
@alan-agius4 Is this new behavior? because when I do `ng update @angular/material`, it only updates Angular Material and the Angular CDK. It doesn't necessarily enforce me to update the CLI too?

alan-agius4 commented:
It was introduced in 8.3.13 actually.

The CLI needs to be updated and be on the same version of the FW.

alan-agius4 commented:
Also as part of the version 9 update we are suggesting that users update to the latest version 8 patch

> Run ng update @angular/core@8 @angular/cli@8 in your workspace directory to update to the latest 8.x version of @angular/core and @angular/cli and commit these changes.

https://update.angular.io/#8.0:9.0

devversion reviewed (approved):
Considering that this logic already existed in version 8, I think it's okay to show the new command.

<sup>Note: My point was mainly: why risk making it not work for some people if we can just keep the existing command that works _always_</sup>

IgorMinar reviewed (approved):
thanks!
